### PR TITLE
Fix delete icon visibility on podcast detail screen

### PR DESCRIPTION
**Pull request by Mzazi25**
## Description

This PR fixes the bookmark icon visibility on the Podcast detail screen as discussed [here](https://github.com/Automattic/pocket-casts-android/pull/1276#issuecomment-1689321629)
The UserEpisodes was resolved on [1276](https://github.com/Automattic/pocket-casts-android/pull/1276)

Fixes # [1273](https://github.com/Automattic/pocket-casts-android/issues/1273)
## Testing Instructions

1. Login using Paid account and sign as Patron
2. Go to Podcast 
3. Play one episode 
4. Add Bookmark 
5. Go to the the Podcast episode list and check the bookmark is added 
6. Delete bookmark. 
7. Observe the Bookmark icon is deleted 

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [X] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt [hunk 1]
```diff
@@ -76,7 +76,11 @@ class EpisodeListAdapter(
                 playbackStateUpdates = playbackManager.playbackStateRelay,
                 upNextChangesObservable = upNextQueue.changesObservable,
                 imageLoader = imageLoader,
-                swipeButtonLayoutFactory = swipeButtonLayoutFactory
+                swipeButtonLayoutFactory = swipeButtonLayoutFactory,
+                userBookmarksObservable = bookmarkManager.findPodcastBookmarksFlow(
+                    podcastUuid = fromListUuid ?: "",
```
Comment (ashiagr) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt:81]: ⚠️ `fromListUuid` is not a `podcastUuid`. We need to get it from the podcast episode.

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt [hunk 2]
```diff
@@ -76,7 +76,11 @@ class EpisodeListAdapter(
                 playbackStateUpdates = playbackManager.playbackStateRelay,
                 upNextChangesObservable = upNextQueue.changesObservable,
                 imageLoader = imageLoader,
-                swipeButtonLayoutFactory = swipeButtonLayoutFactory
+                swipeButtonLayoutFactory = swipeButtonLayoutFactory,
+                userBookmarksObservable = bookmarkManager.findPodcastBookmarksFlow(
```
Comment (ashiagr) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt:80]: Np: Can we rename it to `podcastBookmarksObservable`?

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeViewHolder.kt [hunk 3]
```diff
@@ -166,6 +168,13 @@ class EpisodeViewHolder constructor(
             .startWith(emptyState) // Pre load with a blank state so it doesn't wait for the first update
             .map { if (it.episodeUuid == episode.uuid) it else emptyState } // When another episode is playing return an empty state to clear fields like the buffering status
 
+        data class CombinedData(
+            val downloadProgress: Int,
+            val playbackState: PlaybackState,
+            val isInUpNext: Boolean,
+            val userBookmarks: List<Bookmark>,
```
Comment (ashiagr) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeViewHolder.kt:175]: Np: Can we rename it to `podcastBookmarks`?

**Review by ashiagr — COMMENTED** [review]
Thank you for the contribution, Caleb! 

I left a few nitpicks and a blocker comment. Once those are addressed we're ready to merge this.

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt (hunk 1 above)
Comment (Mzazi25) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt:81]: okay

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt (hunk 2 above)
Comment (Mzazi25) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt:80]: sure

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeViewHolder.kt (hunk 3 above)
Comment (Mzazi25) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeViewHolder.kt:175]: sure

**Comment by Mzazi25** [comment]
@ashiagr  all comments are fixed, You can review now

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt [hunk 4]
```diff
@@ -54,6 +54,8 @@ class EpisodeListAdapter(
     val swipeButtonLayoutFactory: SwipeButtonLayoutFactory,
 ) : ListAdapter<BaseEpisode, RecyclerView.ViewHolder>(PLAYBACK_DIFF) {
 
+    private lateinit var podcastEpisode: PodcastEpisode
```
Comment (ashiagr) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt:57]: ⚠️ `podcastEpisode` is not being set anywhere. Accessing `podcastUuid`  from a null object will cause a crash. Try loading episodes from the filters tab for example.

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt [hunk 5]
```diff
@@ -76,7 +78,11 @@ class EpisodeListAdapter(
                 playbackStateUpdates = playbackManager.playbackStateRelay,
                 upNextChangesObservable = upNextQueue.changesObservable,
                 imageLoader = imageLoader,
-                swipeButtonLayoutFactory = swipeButtonLayoutFactory
+                swipeButtonLayoutFactory = swipeButtonLayoutFactory,
+                podcastBookmarksObservable = bookmarkManager.findPodcastBookmarksFlow(
+                    podcastUuid = podcastEpisode.podcastUuid,
+                    sortType = settings.getBookmarksSortTypeForPodcast()
+                ).asObservable()
```
Comment (ashiagr) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcast/EpisodeListAdapter.kt:85]: How about setting it in the `bindEpisodeViewHolder`  where we have access to the podcast episode?

**Comment by Mzazi25** [comment]
@ashiagr I pushed a commit, it should be good now

**Comment by ashiagr** [comment]
Yay! It works like a charm now. 

Thank you.

**Pull request merged**